spectator-reg-tdigest/src/main/java/com/netflix/spectator/tdigest/StreamTDigestReader.java

```diff
@@ -65,9 +69,10 @@ public StreamTDigestReader(DataInputStream in) {
    * Consume all data from the stream create a list of the measurement batches as they were
    * read from the stream.
    */
-  static List<List<TDigestMeasurement>> readAll(InputStream in) throws IOException {
+  static List<List<TDigestMeasurement>> readAll(Registry registry, InputStream in)
```

Comment:
I think it can change to just a `List<List<TDigestMeasurement>> readAll()` method that uses the stream. I would prefer to do that as a separate PR though.